### PR TITLE
analytics: added new segement events SS-1170

### PR DESCRIPTION
**Pull request by hardik1003**

### CODE REVIEW

File: components/MDXComponents.tsx [hunk 1]
```diff
@@ -41,14 +41,15 @@ const LinkCustom = (props) => {
 
     if (isInternalLink) {
         return (
-            <Link href={href}>
-                <a>{props.children}</a>
+            // @ts-ignore
```
Comment (saikatmitra91) [inline components/MDXComponents.tsx:44]: why ts-ignore ?

File: docs/javascript/v2/debugging/debugging.mdx [hunk 2]
```diff
@@ -56,13 +55,15 @@ on [discord](https://100ms.live/discord), there is a channel there for web-sdk-d
 If you're reporting a bug, we would appreciate as many details possible. We might further ask you for the following information to debug the issues better -
 
 ### Steps to reproduce
-Detailed steps to reproduce the issue. You can also give us a [codesandbox](https://codesandbox.io/) where the issue can be seen, feel free
+
+Detailed steps to reproduce the issue. You can also give us a <a onClick={()=>{window.analytics.track("codesandbox.viewed",{path:window.location.pathname});window.open("https://codesandbox.io","_blank")}}>codesandbox</a> where the issue can be seen, feel free
```
Comment (saikatmitra91) [inline docs/javascript/v2/debugging/debugging.mdx:59]: why not have an override for anchor elements similar to code snippets ? 
why is the logic of firing events exposed here  ? 

we should make adding events simple for people contributing to this repo

**Review by saikatmitra91 — CHANGES_REQUESTED** [review]

File: docs/javascript/v2/debugging/debugging.mdx (hunk 2 above)
Comment (hardik1003) [reply docs/javascript/v2/debugging/debugging.mdx:59]: We cant pass function as well as long string text or string seperated by "."

File: components/MDXComponents.tsx (hunk 1 above)
Comment (hardik1003) [reply components/MDXComponents.tsx:44]: Because then it gives a type error for Next/Link

File: docs/javascript/v2/debugging/debugging.mdx (hunk 2 above)
Comment (saikatmitra91) [reply docs/javascript/v2/debugging/debugging.mdx:59]: we dont need to pass functions right 🤔 ?

**Review by saikatmitra91 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**